### ISSUE TITLE
Questions about the onConnect and Ondisconnect methods

### ISSUE THREAD

**Issue by kexianjun** [apache/dubbo#6544]
when I configure the onconnect or ondisconnect method like following
> @DubboService(onconnect = "onConnect", ondisconnect = "onDisconnect", version = "1.0")

it works on the provider side, but the onconnect or ondisconnect config will propagate to the consumer side, In this scenario,I got exception 
> org.apache.dubbo.remoting.RemotingException: Not found exported service: org.apache.dubbo.demo.DemoService:1.0:20880 in [], may be version or group mismatch , channel: consumer: /192.168.31.94:20880 --> provider: /192.168.31.94:61050, message:RpcInvocation [methodName=onConnect, parameterTypes=[], arguments=[], attachments={path=org.apache.dubbo.demo.DemoService, dubbo.stub.event=true, interface=org.apache.dubbo.demo.DemoService, version=1.0, group=null}]
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocol.getInvoker(DubboProtocol.java:265)
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocol$1.reply(DubboProtocol.java:120)
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocol$1.received(DubboProtocol.java:152)
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocol$1.invoke(DubboProtocol.java:176)
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocol$1.connected(DubboProtocol.java:161)
	at org.apache.dubbo.remoting.exchange.support.header.HeaderExchangeHandler.connected(HeaderExchangeHandler.java:125)
	at org.apache.dubbo.remoting.transport.AbstractChannelHandlerDelegate.connected(AbstractChannelHandlerDelegate.java:43)
	at org.apache.dubbo.remoting.transport.dispatcher.ChannelEventRunnable.run(ChannelEventRunnable.java:66)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

this exception indicate there is not suitable invoker for the connect event. Than I want to make it work for the consumer,I've noticed that `StubProxyFactoryWrapper` had export an invoker in the following code

```java
//export stub service                                                                                                                           
URLBuilder urlBuilder = URLBuilder.from(url);                                                                                                   
if (url.getParameter(STUB_EVENT_KEY, DEFAULT_STUB_EVENT)) {                                                                                     
    urlBuilder.addParameter(STUB_EVENT_METHODS_KEY, StringUtils.join(Wrapper.getWrapper(proxy.getClass()).getDeclaredMethodNames(), ","));      
    urlBuilder.addParameter(IS_SERVER_KEY, Boolean.FALSE.toString());                                                                           
    try {                                                                                                                                       
        export(proxy, (Class) invoker.getInterface(), urlBuilder.build());                                                                      
    } catch (Exception e) {                                                                                                                     
        LOGGER.error("export a stub service error.", e);                                                                                        
    }       
}                                                                                                                                    
```

but when I mad dubbo.stub.event to true, I got the following exception
>java.lang.IllegalStateException: No such extension org.apache.dubbo.rpc.Protocol by name zookeeper
	at org.apache.dubbo.common.extension.ExtensionLoader.findException(ExtensionLoader.java:623)
	at org.apache.dubbo.common.extension.ExtensionLoader.createExtension(ExtensionLoader.java:630)
	at org.apache.dubbo.common.extension.ExtensionLoader.getExtension(ExtensionLoader.java:429)
	at org.apache.dubbo.common.extension.ExtensionLoader.getExtension(ExtensionLoader.java:413)
	at org.apache.dubbo.rpc.Protocol$Adaptive.export(Protocol$Adaptive.java)
	at org.apache.dubbo.rpc.proxy.wrapper.StubProxyFactoryWrapper.export(StubProxyFactoryWrapper.java:123)
	at org.apache.dubbo.rpc.proxy.wrapper.StubProxyFactoryWrapper.getProxy(StubProxyFactoryWrapper.java:93)
	at org.apache.dubbo.rpc.ProxyFactory$Adaptive.getProxy(ProxyFactory$Adaptive.java)
	at org.apache.dubbo.config.ReferenceConfig.createProxy(ReferenceConfig.java:405)
	at org.apache.dubbo.config.ReferenceConfig.init(ReferenceConfig.java:305)
	at org.apache.dubbo.config.ReferenceConfig.get(ReferenceConfig.java:205)
	at org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor.doGetInjectedBean(ReferenceAnnotationBeanPostProcessor.java:144)
	at com.alibaba.spring.beans.factory.annotation.AbstractAnnotationBeanPostProcessor.getInjectedObject(AbstractAnnotationBeanPostProcessor.java:359)
	at com.alibaba.spring.beans.factory.annotation.AbstractAnnotationBeanPostProcessor$AnnotatedFieldElement.inject(AbstractAnnotationBeanPostProcessor.java:539)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at com.alibaba.spring.beans.factory.annotation.AbstractAnnotationBeanPostProcessor.postProcessPropertyValues(AbstractAnnotationBeanPostProcessor.java:142)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1268)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:553)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:312)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:308)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:761)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543)
	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:84)
	at org.apache.dubbo.demo.consumer.Application.main(Application.java:34)

The problem is obvious, the protocol of the url is not right.and also the connect event happened before proxyFactory#getProxy.

And it seems that these two callback methods have only processed in the DubboProtocol.

here is my question:
1. can we prevent these two configs propagate from the provider side to the consumer side?
2. how to make these work on the consumer side?
3. are these two callback methods only work for dubbo protocol？

**Comment by cvictory** [comment]
I think the "onconnect" is used to trigger some action when the connection is created at provider side.
The trigger result will not be transport to consumer side.

If you have not other question, I will close this issue.

**Comment by chickenlj** [comment]
I think they should not be transmitted to Consumer side, we need to fix this in 2.7.10
